plugins/pipelines/create.js

```diff
@@ -48,11 +48,17 @@ module.exports = () => ({
             const userFactory = request.server.app.userFactory;
             const scmUrl = formatScmUrl(request.payload.scmUrl);
             const username = request.auth.credentials.username;
+            let user;
 
             // fetch the user
             return userFactory.get({ username })
                 // get the user permissions for the repo
-                .then(user => user.getPermissions(scmUrl))
+                .then(u => {
+                    // store the user object for fetching token later
+                    user = u;
```

Comment:
I mean you can change it to be in the same scope so that you don't need to store the user object. Something like this: 

```
.then(user => user.getPermissions(scmUrl)
    // user is available in this whole scope
    .then(permissions => { ... })
    .
    .
    .
    .then(() => {
       const admins = {};

       admins[username] = true;

       return user.unsealToken()
           .then(token => options.scmPlugin.getRepoId({ scmUrl, token }))
           .then(repo => { ... })
       })
// this is where it ends
)
```